litestar/dto/msgspec_dto.py

```diff
@@ -40,7 +43,7 @@ def default_or_empty(value: Any) -> Any:
                     field_definition=field_definition,
                     dto_field=dto_field,
                     model_name=model_type.__name__,
-                    default_factory=default_or_empty(msgspec_field.default_factory),
+                    default_factory=default_or_none(msgspec_field.default_factory),
```

Comment:
This was a bug.

The type of `default_factory` is `Callable | None`, so `Empty` was being allowed b/c  as a type it is a callable.